src/XrdTpc/XrdTpcPMarkManager.hh

```diff
@@ -97,16 +109,26 @@ public:
 
   virtual ~PMarkManager() = default;
 private:
+  /**
+  * Add the connected socket information that will be used for packet marking to this manager class
+  * Note: these info will only be added if startTransfer(...) has been called. It allows
+  * to ensure that the connection will be related to the data transfers and not for anything else. We only want
+  * to mark the traffic of the transfers.
+  * @param fd the socket file descriptor
+  * @param sockP the structure describing the address of the socket
+  */
+  void addFd(int fd, const struct sockaddr * sockP);
+
   // The queue of socket information from which we will create the packet marking handles
   std::queue<SocketInfo> mSocketInfos;
   // The map of socket FD and packet marking handles
   std::map<int,std::unique_ptr<XrdNetPMark::Handle>> mPmarkHandles;
   // The instance of the packet marking functionality
   XrdNetPMark * mPmark;
+  // The XrdHttpTPC request information
+  XrdHttpExtReq & mReq;
```

Comment:
There is a potential for a crash here if the reference becomes stale. Is it guaranteed that it will outlive the `PMarkManager`?

Reply:
Yes, because the PMarkManager is a member of the TPCLogRecord which will be destroyed at the end of the transfer, (which will destroy the PMarkManager. But the XrdHttpExtReq will still be alive after the TPCLogRecord instance is destroyed. See: https://github.com/xrootd/xrootd/blob/master/src/XrdTpc/XrdTpcTPC.cc#L300, https://github.com/xrootd/xrootd/blob/master/src/XrdTpc/XrdTpcTPC.cc#L304, https://github.com/xrootd/xrootd/blob/master/src/XrdTpc/XrdTpcTPC.cc#L938 and https://github.com/xrootd/xrootd/blob/master/src/XrdTpc/XrdTpcTPC.cc#L856